ow_dynamic_terrain/src/TerrainModifier.cpp

```diff
@@ -250,9 +252,25 @@ bool TerrainModifier::applyImageToHeightmap(Heightmap* heightmap, const cv::Poin
 
   auto diff = OpenCV_Util::createZerosMatLike(image);
 
+  uint32 mask_size = 0;
+  HardwarePixelBufferSharedPtr pixel_buffer;
+  uint8* mask_pixels = nullptr;
+  if (!maskmap.isNull())
+  {
+    mask_size = maskmap->getWidth();
+    // Lock the mask map pixel buffer
+    pixel_buffer = maskmap->getBuffer();
+    pixel_buffer->lock(HardwareBuffer::HBL_DISCARD);
+    // Get pixel pointer
+    const PixelBox& pixel_box = pixel_buffer->getCurrentLock();
+    mask_pixels = static_cast<uint8*>(pixel_box.data);
+  }
+
```

Comment:
I am not sure what image library you're using here, but I am wondering why it's not OpenCV, like the rest of the code uses? Is it a special case for the mask? Should the other code use it instead of OpenCV?

Reply:
I'm using Ogre API to modify an Ogre texture, which is reasonable. I don't know the specifics of why OpenCV was used to modify an Ogre Heightmap. There might be a good reason for it.